src/store/message-store-level.ts

```diff
@@ -89,7 +100,113 @@ export class MessageStoreLevel implements MessageStore {
       if (message) { messages.push(message); }
     }
 
-    return messages;
+    const sortedRecords = await this.sortMessages(messages, messageSort);
+    return this.paginateRecords(sortedRecords, pagination);
+  }
+
+  private async paginateRecords(
+    messages: GenericMessage[],
+    pagination: Pagination = { }
+  ): Promise<GenericMessage[]> {
+    const { messageCid, limit = 0 } = pagination;
+    if (messageCid === undefined && limit > 0) {
```

Comment:
Sanity: given the earlier than that limit will always be greater than zero, we should probably update the check
```suggestion
    if (messageId === undefined && limit !== undefined) {
```

Reply:
I updated the code according to my comment, see if you like/hate it:


https://github.com/TBD54566975/dwn-sdk-js/pull/478/commits/1125755bcf30e1e961bbe06ea3995dcccbbcd942